main/src/main/java/cgeo/geocaching/backup/BackupActivity.java

```diff
@@ -0,0 +1,67 @@
+package cgeo.geocaching.backup;
+
+import cgeo.geocaching.settings.Settings;
+import cgeo.geocaching.utils.BackupUtils;
+import cgeo.geocaching.utils.ContextLogger;
+import cgeo.geocaching.utils.Log;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+public class BackupActivity extends AppCompatActivity {
+
+    public static final String STATE_BACKUPUTILS = "backuputils";
+
+    private BackupUtils backupUtils = null;
+
+    @Override
+    public void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (Settings.getBackupRestoreRequested() == Settings.BACKUP_NONE) {
+            finish();
+        }
+
+        // prepare restart of c:geo
+        final Intent intent = Settings.getStartscreenIntent(this);
+        intent.putExtras(getIntent());
+
+
+        try (ContextLogger cLog = new ContextLogger(Log.LogLevel.DEBUG, "BackupActivity.onCreate")) {
+            backupUtils = new BackupUtils(this, savedInstanceState == null ? null : savedInstanceState.getBundle(STATE_BACKUPUTILS));
+
+            if ((Settings.getBackupRestoreRequested() & Settings.BACKUP_AUTO) > 0) {
+                Settings.clearBackupRestoreRequested(Settings.BACKUP_AUTO);
+                backupUtils.backup(() -> Settings.setAutomaticBackupLastCheck(false), true);
```

Comment:
Careful. "Backup" also backs up settings. So if executed here and other flags are set (e.g. "RESTORE"), then those would also be backed up. And when this backup then is applied, this flag would also be restored.

Better first check all flags, then clear all flags, then execute depending on the flags.

Reply:
Good catch, will adapt accordingly.

Thanks for taking the time for a review.